internal/store/statefulset_test.go

```diff
@@ -250,14 +261,98 @@ func TestStatefulSetStore(t *testing.T) {
+				# HELP kube_statefulset_status_update_revision [STABLE] Indicates the version of the StatefulSet used to generate Pods in the sequence [replicas-updatedReplicas,replicas)
+				# TYPE kube_statefulset_labels gauge
+				# TYPE kube_statefulset_metadata_generation gauge
+				# TYPE kube_statefulset_persistent_volume_claim_retention_policy gauge
+				# TYPE kube_statefulset_replicas gauge
+				# TYPE kube_statefulset_status_current_revision gauge
+				# TYPE kube_statefulset_status_replicas gauge
+				# TYPE kube_statefulset_status_replicas_available gauge
+				# TYPE kube_statefulset_status_replicas_current gauge
+				# TYPE kube_statefulset_status_replicas_ready gauge
+				# TYPE kube_statefulset_status_replicas_updated gauge
+				# TYPE kube_statefulset_status_update_revision gauge
+				kube_statefulset_status_update_revision{namespace="ns4",revision="ur3",statefulset="statefulset4"} 1
+ 				kube_statefulset_status_replicas{namespace="ns4",statefulset="statefulset4"} 7
+				kube_statefulset_status_replicas_available{namespace="ns4",statefulset="statefulset4"} 0
+				kube_statefulset_status_replicas_current{namespace="ns4",statefulset="statefulset4"} 0
+				kube_statefulset_status_replicas_ready{namespace="ns4",statefulset="statefulset4"} 0
+				kube_statefulset_status_replicas_updated{namespace="ns4",statefulset="statefulset4"} 0
+ 				kube_statefulset_replicas{namespace="ns4",statefulset="statefulset4"} 3
+ 				kube_statefulset_metadata_generation{namespace="ns4",statefulset="statefulset4"} 1
+ 				kube_statefulset_persistent_volume_claim_retention_policy{namespace="ns4",statefulset="statefulset4",when_deleted="Retain",when_scaled="Delete"} 1
+				kube_statefulset_labels{namespace="ns4",statefulset="statefulset4"} 1
+				kube_statefulset_status_current_revision{namespace="ns4",revision="cr3",statefulset="statefulset4"} 1
+ 			`,
+			MetricNames: []string{
+				"kube_statefulset_labels",
+				"kube_statefulset_metadata_generation",
+				"kube_statefulset_replicas",
+				"kube_statefulset_status_replicas",
+				"kube_statefulset_status_replicas_available",
+				"kube_statefulset_status_replicas_current",
+				"kube_statefulset_status_replicas_ready",
+				"kube_statefulset_status_replicas_updated",
+				"kube_statefulset_status_update_revision",
+				"kube_statefulset_status_current_revision",
+				"kube_statefulset_persistent_volume_claim_retention_policy",
 			},
 		},
 	}
 	for i, c := range cases {
 		c.Func = generator.ComposeMetricGenFuncs(statefulSetMetricFamilies(nil, nil))
 		c.Headers = generator.ExtractMetricFamilyHeaders(statefulSetMetricFamilies(nil, nil))
+		ord := "th"
+		switch i + 1 {
+		case 1:
+			ord = "st"
+		case 2:
+			ord = "nd"
+		case 3:
+			ord = "rd"
+		}
 		if err := c.run(); err != nil {
-			t.Errorf("unexpected collecting result in %vth run:\n%s", i, err)
+			t.Errorf("unexpected collecting result in %v%s run:\n%s", i+1, ord, err)
```

Comment:
```suggestion
			t.Errorf("unexpected collecting result in run %v:\n%s" i, err)
```

Let's reduce complexity here and rephrase the error message instead

Reply:
Done, note that we need to use i+1 to have the index match the names used in the test cases (which is the practical way to figure out which test failed).